ocaml/ocaml
# Unified GitHub Actions workflows across release branches for ease-of-maintenance [4.14 version]

See #12846 for the motivation for this series of PRs. 

This PR deletes `.github/workflows/stale.yml` as it is not needed. The updates in `manual/` and `testsuite/` are backported from #11346 and #11617. The remaining changes are solely in `.github/workflows` and `tools/ci/actions`. The difference in these files between the content of #12846 and this PR is the result of `git diff ce1c3e14b7 596646270c -- .github/workflows tools/ci/actions` which is:

```diff
diff --git a/.github/workflows/build.yml b/.github/workflows/build.yml
index 03d95cc51d..2736f4be2b 100644
--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -11,12 +11,6 @@ on:
 # Restrict the GITHUB_TOKEN
 permissions: {}
 
-# List of test directories for the debug-s4096 and linux-O0 jobs.
-# These directories are selected because of their tendencies to reach corner
-# cases in the runtime system.
-env:
-  PARALLEL_TESTS: parallel callback gc-roots weak-ephe-final
-
 # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency
 # Concurrent workflows are grouped by the PR or branch that triggered them
 # (github.ref) and the name of the workflow (github.workflow). The
@@ -74,8 +68,6 @@ jobs:
 # "extra" testsuite runs, reusing the previously built compiler tree.
 # debug: running the full testsuite with the
 #        debug runtime and minor heap verification.
-# debug-s4096: select testsuite run with the debug runtime and a small
-#              minor heap.
   normal:
     name: ${{ matrix.name }}
     needs: build
@@ -88,8 +80,6 @@ jobs:
             dependencies: texlive-latex-extra texlive-fonts-recommended hevea sass
           - id: debug
             name: extra (debug)
-          - id: debug-s4096
-            name: extra (debug-s4096)
     steps:
       - name: Download Artifact
         uses: actions/download-artifact@v3
@@ -114,15 +104,6 @@ jobs:
           USE_RUNTIME: d
         run: |
           bash -cxe "SHOW_TIMINGS=1 tools/ci/actions/runner.sh test"
-      - name: Run the testsuite (s=4096, debug runtime)
-        if: matrix.id == 'debug-s4096'
-        env:
-          OCAMLRUNPARAM: s=4096,v=0
-          USE_RUNTIME: d
-        run: |
-          for dir in $PARALLEL_TESTS; do \
-            bash -cxe "SHOW_TIMINGS=1 tools/ci/actions/runner.sh test_prefix $dir"; \
-          done
       - name: Build API Documentation
         if: matrix.id == 'normal'
         run: |
@@ -140,16 +121,18 @@ jobs:
         run: |
           MAKE_ARG=-j OCAMLRUNPARAM=b,v=0 bash -xe tools/ci/actions/runner.sh other-checks
 
-# MacOS build+testsuite run, and Linux O0 run.
+# MacOS build+testsuite run, and no-naked-pointers run.
   others:
     name: ${{ matrix.name }}
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         include:
-          - name: linux-O0
+          - name: no-naked-pointers
             os: ubuntu-latest
-            config_arg: CFLAGS='-O0'
+            config_arg: >-
+              --disable-naked-pointers
+              --disable-stdlib-manpages
           - name: macos
             os: macos-latest
     steps:
@@ -170,20 +153,11 @@ jobs:
         if: ${{ matrix.name != 'linux-O0' }}
         run: |
           bash -c 'SHOW_TIMINGS=1 tools/ci/actions/runner.sh test'
-      - name: Run the testsuite (linux-O0)
-        if: ${{ matrix.name == 'linux-O0' }}
-        env:
-          OCAMLRUNPARAM: v=0,V=1
-          USE_RUNTIME: d
-        run: |
-          for dir in $PARALLEL_TESTS; do \
-           bash -cxe "SHOW_TIMINGS=1 tools/ci/actions/runner.sh test_prefix $dir"; \
-          done
 
   i386:
     runs-on: ubuntu-latest
     container:
-      image: debian:12
+      image: debian:10
       options: --platform linux/i386 --user root
     steps:
       - name: OS Dependencies
diff --git a/.github/workflows/hygiene.yml b/.github/workflows/hygiene.yml
index 777a6b4e5c..269c170f9b 100644
--- a/.github/workflows/hygiene.yml
+++ b/.github/workflows/hygiene.yml
@@ -28,6 +28,10 @@ jobs:
         # context variable.
         if: failure()
 
+      - name: Packages
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y autoconf2.69
+
       - uses: actions/checkout@v3
         with:
           fetch-depth: 50
diff --git a/tools/ci/actions/check-configure.sh b/tools/ci/actions/check-configure.sh
index 44e1452a54..743fca2611 100755
--- a/tools/ci/actions/check-configure.sh
+++ b/tools/ci/actions/check-configure.sh
@@ -69,7 +69,7 @@ CheckTree () {
     git checkout -qB return
     git checkout -q "$COMMIT"
     mv configure configure.ref
-    make -s configure
+    make -s configure AUTOCONF_TOOL_NAME=autoconf2.69
     if diff -q configure configure.ref >/dev/null ; then
       echo -e "$COMMIT: \e[32mconfigure.ac generates configure\e[0m"
     else
diff --git a/tools/ci/actions/runner.sh b/tools/ci/actions/runner.sh
index b22ec0eafa..fc77609061 100755
--- a/tools/ci/actions/runner.sh
+++ b/tools/ci/actions/runner.sh
@@ -60,7 +60,7 @@ Build () {
   fi
   rm build.log
   echo Ensuring that all names are prefixed in the runtime
-  if ! ./tools/check-symbol-names runtime/*.a otherlibs/*/lib*.a ; then
+  if ! ./tools/check-symbol-names runtime/*.a ; then
     failed=1
   fi
   if ((failed)); then
@@ -74,15 +74,6 @@ Test () {
   cd ..
 }
 
-# By default, TestPrefix will attempt to run the tests
-# in the given directory in parallel.
-TestPrefix () {
-  TO_RUN=parallel-"$1"
-  echo Running single testsuite directory with $TO_RUN
-  $MAKE -C testsuite $TO_RUN
-  cd ..
-}
-
 API_Docs () {
   echo Ensuring that all library documentation compiles
   $MAKE -C api_docgen html pdf texi
@@ -168,7 +159,6 @@ case $1 in
 configure) Configure;;
 build) Build;;
 test) Test;;
-test_prefix) TestPrefix $2;;
 api-docs) API_Docs;;
 install) Install;;
 manual) BuildManual;;
```
